File: SDKs/amazonaws.com/mgn/2020-02-26/python/src/sdk/models/shared/datareplicationinfo.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import datareplicationerror as shared_datareplicationerror
from ..shared import datareplicationinitiation as shared_datareplicationinitiation
from ..shared import datareplicationstate_enum as shared_datareplicationstate_enum
from ..shared import datareplicationinforeplicateddisk as shared_datareplicationinforeplicateddisk


@dataclass_json
@dataclasses.dataclass
class DataReplicationInfo:
    r"""DataReplicationInfo
    Request data replication info.
    """
    
    data_replication_error: Optional[shared_datareplicationerror.DataReplicationError] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dataReplicationError') }})
    data_replication_initiation: Optional[shared_datareplicationinitiation.DataReplicationInitiation] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dataReplicationInitiation') }})
    data_replication_state: Optional[shared_datareplicationstate_enum.DataReplicationStateEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dataReplicationState') }})
    eta_date_time: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('etaDateTime') }})
    lag_duration: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lagDuration') }})
    replicated_disks: Optional[list[shared_datareplicationinforeplicateddisk.DataReplicationInfoReplicatedDisk]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('replicatedDisks') }})
    

File: SDKs/amazonaws.com/mgn/2020-02-26/java/lib/src/main/java/openapisdk/models/shared/DataReplicationInfo.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * DataReplicationInfo
 * Request data replication info.
**/
public class DataReplicationInfo {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("dataReplicationError")
    public DataReplicationError dataReplicationError;
    public DataReplicationInfo withDataReplicationError(DataReplicationError dataReplicationError) {
        this.dataReplicationError = dataReplicationError;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("dataReplicationInitiation")
    public DataReplicationInitiation dataReplicationInitiation;
    public DataReplicationInfo withDataReplicationInitiation(DataReplicationInitiation dataReplicationInitiation) {
        this.dataReplicationInitiation = dataReplicationInitiation;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("dataReplicationState")
    public DataReplicationStateEnum dataReplicationState;
    public DataReplicationInfo withDataReplicationState(DataReplicationStateEnum dataReplicationState) {
        this.dataReplicationState = dataReplicationState;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("etaDateTime")
    public String etaDateTime;
    public DataReplicationInfo withEtaDateTime(String etaDateTime) {
        this.etaDateTime = etaDateTime;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("lagDuration")
    public String lagDuration;
    public DataReplicationInfo withLagDuration(String lagDuration) {
        this.lagDuration = lagDuration;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("replicatedDisks")
    public DataReplicationInfoReplicatedDisk[] replicatedDisks;
    public DataReplicationInfo withReplicatedDisks(DataReplicationInfoReplicatedDisk[] replicatedDisks) {
        this.replicatedDisks = replicatedDisks;
        return this;
    }
}